src/aioquic/h3/connection.py

```diff
@@ -194,19 +200,46 @@ def stream_is_request_response(stream_id: int):
     return stream_id % 4 == 0
 
 
+def validate_key(key: bytes):
+    # RFC 9113 section 8.2.1
+    for i, c in enumerate(key):
+        if c <= 0x20 or (c >= 0x41 and c <= 0x5A) or c >= 0x7F:
+            raise MessageError("Header %r contains invalid characters" % key)
+        if c == COLON and i != 0:
+            # Colon not at start, definitely bad.  Keys starting with a colon
+            # will be checked in pseudo-header validation code.
+            raise MessageError("Header %r contains a non-initial colon" % key)
+
+
+def validate_value(key: bytes, value: bytes):
+    # RFC 9113 section 8.2.1
+    for c in value:
+        if c == NUL or c == LF or c == CR:
+            raise MessageError("Header %r value has forbidden characters" % key)
+    if len(value) > 0:
+        first = value[0]
+        if first == SP or first == TAB:
```

Comment:
Are CR or LF whitespace too?

Reply:
CR and LF (along with NUL) are forbidden anywhere in the value by the RFC whereas SP and TAB are only banned as initial or final characters.  I contemplated making WHITESPACE(x) == (x == SP or x == TAB) but didn't as I in other situations WHITESPACE often includes CR and LF, and I wanted to make the translation from the RFC as literal as possible.  But I'm happy with making a WHITESPACE if you prefer it.

Reply:
I'd be fine with `WHITESPACE = (CR, LF) ` for instance, with a comment explaning why we don't need to test other sorts of whitespace.